source/includes/_endpoint-dataset.md

```diff
@@ -565,3 +565,23 @@ PATCH the "expression" attribute to modify. An empty "expression" object, like
 ##### Decks
 
 ##### Permissions
+
+##### Primary key
+
+`/datasets/{id}/pk`
+
+Setting a primary key on a dataset causes updates (particularly streamed
+updates) mentioning existing rows to be updated instead of new rows being
+inserted.
+
+GET on this resource returns a Shoji Entity with a "pk"" attribute in its body,
+which is a list. The "pk" attribute indicates the variable ids of the columns in
```

Comment:
Convo in slack about this:

```
npr: if you're still computering, i'm trying to wrap my head around the concept of possibly returning a URL from the 
entity controller for a pk... here's the code...
 https://github.com/Crunch-io/backend-http/blob/streamed-updates-111713740/cr/server/api/controllers/datasets/entity.py#L127

[12:33] 
i dont think we want to return a URL, because the pk value itself can't be traversed to

npr [12:34 AM] 
Is it not an array of variable ids?

mcdonc [12:34 AM] 
it is, but it doesnt form a collection of any kind, it's just a value

npr [12:35 AM] 
We typically urlize ids when they come out of the http api. Just suggesting that we follow what I understand our 
convention to be. Perhaps you/we can discuss with bob tomorrow

mcdonc [12:37 AM] 
ok... for posterity and others to see tomorrow, the pk value is indeed a sequence of variable ids, but none of those
variable ids has any attributes of their own.. its the end of the line

npr [12:37 AM] 
As someone who has written a lot of our Api docs, I hate when I have to explain why this endpoint works one way 
but this other seemingly similar endpoint is different for no reason other than that a different developer wrote it. And 
it's a bitch to write a client for such an app

[12:37] 
*api

mcdonc [12:38 AM] 
certainly

[12:38] 
and that's definitely the case here, i'm fucken clueless :slightly_smiling_face:

npr [12:38 AM] 
Ok. We can appeal to the bob for enlightenment tomorrow
```